src/terms.ml

```diff
@@ -12,6 +12,20 @@ open Timed
 
 (** {3 Term (and symbol) representation} *)
 
+(** Symbol properties. *)
+type prop =
+  | Constant
+  (** The symbol is constant. *)
+  | Injective
+  (** The symbol is injective. *)
+
+(** Exposition tags. *)
+type expo =
+  | Protected
+  (** Exposed but usable in LHS arguments only. *)
+  | Private
+  (** Not visible outside of its module. *)
```

Comment:
It is a really bad idea not to include the "normal" case in these types. Going through an indirection with the option type induces extra pointer dereferencing everywhere. Just add constructors `Defin` and `Public`, respectively. Also, I would stick with shorter names to avoid longer lines in the code. The names do not have to exactly match what is in the syntax.

Reply:
Should be fixed in 053cc3646c21fa4000d6ddfca1ab1b02f257994e .